File: Python/robot_turn_2.py
import serial
import time

def robot_turn_2 (s, currentj1, currentj36):
	j1 = -(currentj1 - 900)
	j36 = -(currentj36 + 400)

	turnj1   = '@STEP 221,' + str(j1) + ',0,0,0,0,0,0'
	recoj1   = '@STEP 221,1050,0,0,0,0,0,0'
	turnj2   = '@STEP 221,0,510,0,0,0,0,0'
	turnj2i  = '@STEP 221,0,-510,0,0,0,0,0'
	recoj2   = '@STEP 221,0,700,0,0,0,0,0'
	recoj2i  = '@STEP 221,0,-700,0,0,0,0,0'
	oj36     = '@STEP 221,0,0,' + str(j36) + ',0,0,' + str(j36) + ',0'
	turnj36i = '@STEP 221,0,0,380,0,0,380,0'
	turnj45  = '@STEP 221,0,0,0,-340,-340,0,0'
	rotaj45  = '@STEP 221,0,0,0,-770,770,0,0'
	gp_op    = '@STEP 221,0,0,0,0,0,110,0'
	gp_cl    = '@STEP 221,0,0,0,0,0,-110,0'

	for k in range(16):
	    if k == 0:
	        s.write(turnj1)
	    elif k == 1:
	        s.write(oj36)
	    elif k == 2:
	        s.write(turnj2)
	        time.sleep(3.5)
	    elif k == 3:
	        s.write(gp_op)
	        time.sleep(1.5)
	    elif k == 4:
	        s.write(turnj2i)
	        time.sleep(3.5)
	    elif k == 5:
	        s.write(turnj36i)
	        time.sleep(3);
	    elif k == 6:
	        s.write(turnj45)
	        time.sleep(2.5)
	    elif k == 7:
	        s.write(recoj2)
	        time.sleep(4.5)
	    elif k == 8:
	        s.write(gp_cl)
	        time.sleep(1.5)
	    elif k == 9:
	        s.write(recoj2i)
	        time.sleep(4.5)
	    elif k == 10:
	        s.write(rotaj45)
	        time.sleep(4.5)
	    elif k == 11:
	        s.write(recoj2)
	        time.sleep(4.5)
	    elif k == 12:
	        s.write(gp_op)
	        time.sleep(1.5)
	    elif k == 13:
	        s.write(gp_cl)
	        time.sleep(1.5)
	    elif k == 14:
	        s.write(recoj2i)
	        time.sleep(4.5)
	    else:
	        s.write(recoj1)
	        time.sleep(6)

	    if k == 1:
	        if abs(j1) < 1000:
	            time.sleep(5.5)
	        else:
	            u1 = 0.5*(round((abs(j1)-1000),-1)/100) + 5.5
	            time.sleep(u1)
	    elif k == 2:
	        if abs(j36) <= 200:
	            time.sleep(2)
	        else:
	            u2 = 0.5*(round((abs(j36)-200),-1)/100) + 2
	            time.sleep(u2)
	currentj1 = 1050
	currentj36 = -20
	return currentj1, currentj36

File: Matlab/old/新增資料夾/robot_turn_2.m
function [currentj1,currentj36] =robot_turn_2(s,currentj1,currentj36)
j1 = -(currentj1 - 900);
j36 = -(currentj36 + 400);
%---------------------------robot action-------------------------------------
turnj1 = ['@STEP 221,',num2str(j1),',0,0,0,0,0,0'];
recoj1 = ['@STEP 221,1050,0,0,0,0,0,0'];

turnj2 = ['@STEP 221,0,510,0,0,0,0,0'];
turnj2i = ['@STEP 221,0,-510,0,0,0,0,0'];
recoj2 = ['@STEP 221,0,700,0,0,0,0,0'];
recoj2i = ['@STEP 221,0,-700,0,0,0,0,0'];

oj36 = ['@STEP 221,0,0,',num2str(j36),',0,0,',num2str(j36),',0'];
turnj36i = ['@STEP 221,0,0,380,0,0,380,0'];

turnj45 = ['@STEP 221,0,0,0,-340,-340,0,0'];
rotaj45 = ['@STEP 221,0,0,0,-770,770,0,0'];

gp_op = ['@STEP 221,0,0,0,0,0,110,0'];
gp_cl = ['@STEP 221,0,0,0,0,0,-110,0'];

%---------------------------robot run-------------------------------------

for k = 1:16
    if k == 1
        fprintf(s,turnj1);
    elseif k == 2
        fprintf(s,oj36);
    elseif k == 3
        fprintf(s,turnj2);
        pause(3.5);
    elseif k == 4
        fprintf(s,gp_op);
        pause(1.5);
    elseif k == 5
        fprintf(s,turnj2i);
        pause(3.5);
    elseif k == 6
        fprintf(s,turnj36i);
        pause(3);
    elseif k == 7
        fprintf(s,turnj45);
        pause(2.5);
    elseif k == 8
        fprintf(s,recoj2);
        pause(4.5);
    elseif k == 9
        fprintf(s,gp_cl);
        pause(1.5);
    elseif k == 10
        fprintf(s,recoj2i);
        pause(4.5);
    elseif k == 11
        fprintf(s,rotaj45);
        pause(4.5);
    elseif k == 12
        fprintf(s,recoj2);
        pause(4.5);
    elseif k == 13
        fprintf(s,gp_op);
        pause(1.5);
    elseif k == 14
        fprintf(s,gp_cl);
        pause(1.5);
    elseif k == 15
        fprintf(s,recoj2i);
        pause(4.5);
    else
        fprintf(s,recoj1);
        pause(6);
    end
    %------------------------------------
    if k == 1
        if j1 < 1000
            pause(5.5);
        else
            u1 = 0.5*(round((j1-1000),-1)/100) + 5.5;
            pause(u1);
        end
    elseif k == 2
        if (-j36) <= 200
            pause(2);
        else
            u2 = 0.5*(round((-j36-200),-1)/100) + 2;
            pause(u2);
        end
    end
end
currentj1 = 1050;
currentj36 = -20;
end
%}